src/BaseFirestoreRepository.ts

```diff
@@ -50,38 +50,56 @@ export class BaseFirestoreRepository<T extends IEntity>
   }
 
   async create(item: T): Promise<T> {
-    if (item.id) {
-      const found = await this.findById(item.id);
-      if (found) {
-        return Promise.reject(
-          new Error(`A document with id ${item.id} already exists.`)
-        );
+    try {
+      if (item.id) {
+        const found = await this.findById(item.id);
+        if (found) {
+          throw `A document with id ${item.id} already exists.`;
+        }
+      };
+       
+      const errors = await this.validate(item);
+
+      if (errors.length) {
+        throw errors;
```

Comment:
Yeah good idea. I'll change the order too

Reply:
I looked at adding in custom errors. Couldn't seem to get it to work (could be a bug in chai, or my knowledge simply being lacking), but I have added in an additional check on the message so that it only checks for the validation errors.

Reply:
No problems! The abstraction of the errors is something that is mentioned in the roadmap anyway, I'll check it after.